pandas/core/indexes/interval.py

```diff
@@ -799,29 +799,22 @@ def length(self):
         return Index(self._data.length, copy=False)
 
     def putmask(self, mask, value):
-        arr = self._data.copy()
+        mask = np.asarray(mask, dtype=bool)
```

Comment:
can't you share / use the array/interval putmask code here?

Reply:
we do; L814-815 is directly using IntervalArray.putmask.  everything before that point is for ways that the Index method behaves differently from the array method